pkg/common/resources/quantity.go

```diff
@@ -0,0 +1,103 @@
+/*
+ Licensed to the Apache Software Foundation (ASF) under one
+ or more contributor license agreements.  See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership.  The ASF licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package resources
+
+import (
+	"errors"
+	"math/big"
+	"regexp"
+	"strconv"
+)
+
+// This code handles parsing of SI units in quantities:
+// <quantity>     ::= <signedNumber><suffix>
+// <signedNumber> ::= <digits> | <sign><digits>
+// <sign>         ::= "+" | "-"
+// <digit>        ::= 0 | 1 | ... | 9
+// <digits>       ::= <digit> | <digit><digits>
+// <suffix>       ::= <binarySI> | <decimalSI>
+// <binarySI>     ::= Ki | Mi | Gi | Ti | Pi | Ei
+// <decimalSI>    ::= "" | k | K | M | G | T | P | E
+// Additionally, ParseVCore supports decimalSI of 'm' to indicate millicore.
+
+var whitespace = regexp.MustCompile(`\s+`)
```

Comment:
Use a `strings.TrimSpace()` instead of a regexp. We do not want to allow "1 000 000" as a value which we do with the current setup. If we allow spaces between value and suffix we need to add it in the regexp as below.

Reply:
Done.